odl/solvers/advanced/proximal_operators.py

```diff
@@ -124,14 +133,180 @@ def cconj_prox_factory(step_size):
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_.
+    """
+
+    if a == 0:
+        raise ValueError('scaling parameter {} not non-zero'.format(a))
+
+    def arg_scaling_prox_factory(step_size):
+        """Create proximal for the translation with a given step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * F( . * 1/a)`` where ``s`` is the
+            step size
+        """
+        the_prox = prox_factory(step_size / a**2)
+        return a * the_prox * ScalingOperator(the_prox.domain, 1 / a)
+
+    return arg_scaling_prox_factory
+
+
+def proximal_quadratic_perturbation(prox_factory, a, u):
+    """Calculate the proximal of function F(x) + a/2 * ||x||^2 + <u,x>.
+
+    This is calculated according to the rule
+
+        prox[s * (F( . ) + a/2 * || . ||^2 + <u, . >)](x) =
+        1/sqrt(s*a + 1) prox[s*f( . * 1/sqrt(s*a + 1))]((x - s*u)/sqrt(s*a+1))
+
+    where ``a`` is the scaling parameter belonging to the quadratic term, and
+    ``u`` is the vector defining the linear functional.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    a : non-negative `float`
+        Scaling of the quadratic term
+    u : Element in domain of F
+        Defines the linear functional
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [ComPes2011]_. Note that this
+    identity is not the exact one given, but recalculated for arbitrary step
+    lengths.
+    """
+
+    if a < 0:
+        raise ValueError('scaling parameter {} not non-negative'.format(a))
+
+    if not isinstance(u, LinearSpaceVector):
+        raise TypeError('vector {!r} not a LinearSpaceVector instance.'
+                        ''.format(u))
+
+    if not u.space == prox_factory(1).domain:
+        raise TypeError('vector {!r} not in the domain of the proximal'
+                        'operator {}'.format(u, prox_factory(1).domain))
+
+    def quadratic_perturbation_prox_factory(step_size):
+        """Create proximal for the quadratic perturbation with a given
+        step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * (F(x) + a/2 * ||x||^2 + <u,x>)``,
+            where ``s`` is the step size
+        """
+        const = 1.0 / sqrt(step_size * a + 1)
+        the_prox = proximal_arg_scaling(prox_factory, 1.0 / const)(step_size)
```

Comment:
This does not seem to agree with your text, since the defintion of `proximal_arg_scaling` is `F(x / a)` this would be equivalent to `F(x * sqrt( ... ))`, but in the doc you say `F(x / sqrt( ... ))`. This would be fixed if the definition of `proximal_arg_scaling` was changed.